### PR TITLE
Build kererberized_hadoop image by downloading commons-daemon via https

### PR DESCRIPTION
**Pull request by ilejn**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Addresses https://github.com/ClickHouse/ClickHouse/issues/62873

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 3519a5f0f984ae60682cee33a955e73fe342755b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62886/3519a5f0f984ae60682cee33a955e73fe342755b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by ilejn** [comment]
Hello @Felixoid , could you have a look , please?

Could you explain what was the point of the change https://github.com/ClickHouse/ClickHouse/pull/34556/commits/0ae948866a1105569176e30b4e8667270ff9b656 ?
Effectively is masks errors and makes any analysis more complex.

**Review by Felixoid — COMMENTED** [review]
Unfortunately, I don't remember what was broken in the tests two years ago. Most probably, some of our tests assumed we have Hadoop in the `/usr/local/hadoop` directory.

I can't say detailed, it was somehow broken before the changes, and fixed after

**Comment by Felixoid** [comment]
I think, you mean that `||` works on ANY error. It's a bug, it should be `{ [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop; }`

File: docker/test/integration/kerberized_hadoop/Dockerfile [hunk 1]
```diff
@@ -15,7 +15,10 @@ RUN curl -o krb5-libs-1.10.3-65.el6.x86_64.rpm ftp://ftp.pbone.net/mirror/vault.
     rm -fr *.rpm
 
 RUN cd /tmp && \
-    curl http://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz -o commons-daemon-1.0.15-src.tar.gz && \
+    curl -o wget.rpm ftp://ftp.pbone.net/mirror/vault.centos.org/6.9/os/x86_64/Packages/wget-1.12-10.el6.x86_64.rpm && \
+    rpm -i wget.rpm && \
+    rm -fr *.rpm && \
+    wget --no-check-certificate https://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz && \
```
[the remark this answers is not in the document]
Comment (Felixoid) [reply docker/test/integration/kerberized_hadoop/Dockerfile:21]:
```diff
diff --git a/docker/test/integration/kerberized_hadoop/Dockerfile b/docker/test/integration/kerberized_hadoop/Dockerfile
index 592c3e36ef7..74f60319314 100644
--- a/docker/test/integration/kerberized_hadoop/Dockerfile
+++ b/docker/test/integration/kerberized_hadoop/Dockerfile
@@ -14,11 +14,12 @@ RUN curl -o krb5-libs-1.10.3-65.el6.x86_64.rpm ftp://ftp.pbone.net/mirror/vault.
     rpm -Uvh libkadm5-1.10.3-65.el6.x86_64.rpm libss-1.41.12-24.el6.x86_64.rpm krb5-libs-1.10.3-65.el6.x86_64.rpm krb5-workstation-1.10.3-65.el6.x86_64.rpm libcom_err-1.41.12-24.el6.x86_64.rpm && \
     rm -fr *.rpm

+ADD https://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz /tmp/commons-daemon-1.0.15-src.tar.gz
+
 RUN cd /tmp && \
-    curl http://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz -o commons-daemon-1.0.15-src.tar.gz && \
     tar xzf commons-daemon-1.0.15-src.tar.gz && \
     cd commons-daemon-1.0.15-src/src/native/unix && \
     ./configure && \
     make && \
     cp ./jsvc /usr/local/hadoop-2.7.0/sbin && \
-    [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop
+    { [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop; }
```

Let's not use any of them! =)

[the remark this answers is not in the document]
Comment (Felixoid) [reply docker/test/integration/kerberized_hadoop/Dockerfile:21]: and add cleaning up for `/tmp/common-daemon` in the same `RUN`, please

```
> docker run -it kerberized-hadoop ls /tmp -l
total 300
drwxr-xr-x 3 root root     21 May 16  2015 Jetty_0_0_0_0_50070_hdfs____w2cu08
drwxr-xr-x 3 root root     21 May 16  2015 Jetty_0_0_0_0_50090_secondary____y6aanv
drwxr-xr-x 3 root root     21 May 16  2015 Jetty_localhost_57825_datanode____.7653pc
drwxr-xr-x 3 root root     21 May 16  2015 Jetty_localhost_57908_datanode____jxcs5i
drwxr-xr-x 3 root root    177 Apr 24 12:02 commons-daemon-1.0.15-src
-rw------- 1 root root 291128 Apr  2  2013 commons-daemon-1.0.15-src.tar.gz
drwxr-xr-x 1 root root     17 May 16  2015 hadoop-root
-rw-r--r-- 1 root root      4 May 16  2015 hadoop-root-datanode.pid
-rw-r--r-- 1 root root      4 May 16  2015 hadoop-root-namenode.pid
-rw-r--r-- 1 root root      4 May 16  2015 hadoop-root-secondarynamenode.pid
drwxr-xr-x 1 root root     72 May 16  2015 hsperfdata_root
```

[the remark this answers is not in the document]
Comment (Felixoid) [reply docker/test/integration/kerberized_hadoop/Dockerfile:21]: And answering 

> Shouldn't we replace
> 
> ```
>     ...
>     cp ./jsvc /usr/local/hadoop-2.7.0/sbin && \
>     [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop
> ```
> 
> by
> 
> ```
>     ...
>     cp ./jsvc /usr/local/hadoop-2.7.0/sbin && \
>     ([ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop)
> ```
> 
> ?

the diff works:

```
=> ERROR [4/4] RUN cd /tmp &&     curl -k https://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz -o commons-daemon-1.0  266.9s
------
 > [4/4] RUN cd /tmp &&     curl -k https://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz -o commons-daemon-1.0.15-src.tar.gz &&     tar xzf commons-daemon-1.0.15-src.tar.gz &&     cd commons-daemon-1.0.15-src/src/native/unix &&     ./configure &&     make &&     cp ./jsvc /usr/local/hadoop-2.7.0/sbin &&     { [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop; }:
133.8   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
133.8                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:02:13 --:--:--     0
266.8 curl: (35) SSL connect error
------
Dockerfile:17
--------------------
  16 |
  17 | >>> RUN cd /tmp && \
  18 | >>>     curl -k https://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz -o commons-daemon-1.0.15-src.tar.gz && \
  19 | >>>     tar xzf commons-daemon-1.0.15-src.tar.gz && \
  20 | >>>     cd commons-daemon-1.0.15-src/src/native/unix && \
  21 | >>>     ./configure && \
  22 | >>>     make && \
  23 | >>>     cp ./jsvc /usr/local/hadoop-2.7.0/sbin && \
  24 | >>>     { [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop; }
  25 |
--------------------
ERROR: failed to solve: process "/bin/sh -c cd /tmp &&     curl -k https://archive.apache.org/dist/commons/daemon/source/commons-daemon-1.0.15-src.tar.gz -o commons-daemon-1.0.15-src.tar.gz &&     tar xzf commons-daemon-1.0.15-src.tar.gz &&     cd commons-daemon-1.0.15-src/src/native/unix &&     ./configure &&     make &&     cp ./jsvc /usr/local/hadoop-2.7.0/sbin &&     { [ -e /usr/local/hadoop ] || ln -s ./hadoop-2.7.0 /usr/local/hadoop; }" did not complete successfully: exit code: 35
```

[the remark this answers is not in the document]
Comment (ilejn) [reply docker/test/integration/kerberized_hadoop/Dockerfile:21]: Not sure that I understand ... you suggest using curly brackets? Why? 
They are used in bash for some special purposes, see https://www.linux.com/topic/desktop/all-about-curly-braces-bash/

[the remark this answers is not in the document]
Comment (Felixoid) [reply docker/test/integration/kerberized_hadoop/Dockerfile:21]: Don't mess the curly-brace expansions with commands combination.

`{}` is the same as `()`, but the latter spawns another shell instance, and the first doesn't. They are almost equal, with a tiny difference. The first is a bit more resource-saving. I avoid using `()` when it's possible


```
$ true && ( echo $$; echo $BASHPID; false || false; ); echo $? $$ $BASHPID
620964
621744
1 620964 620964
$ true && { echo $$; echo $BASHPID; false || false; }; echo $? $$ $BASHPID
620964
620964
1 620964 620964
$ true && ( echo $$; echo $BASHPID; true || false; ); echo $? $$ $BASHPID
620964
621842
0 620964 620964
$ true && { echo $$; echo $BASHPID; true || false; }; echo $? $$ $BASHPID
620964
620964
0 620964 620964
```

**Comment by ilejn** [comment]
Error "TypeError: 'bool' object is not callable" in ci.py seems to be persistent.
I doubt that it is caused by the nature of the change.

**Pull request merged**